Python/Python.sublime-syntax

```diff
@@ -1016,8 +1016,17 @@ contexts:
 
   assignments:
     - include: illegal-assignment-expression
-    - match: ':'
-      scope: punctuation.separator.annotation.variable.python
+    - match: '(?=:)'
+      set:
+        - match: ':'
+          scope: punctuation.separator.annotation.variable.python
+          set:
+            - meta_scope: meta.variable.parameters.annotation.python
```

Comment:
If we expect annotations to be primarily used for type-hints existing `meta.function.parameters.annotation` is not very suiteable anymore. The concept of `parameters` in assignments feels even less appropriate, IMHO.

`meta.annotation` would generally be a good scope, but it is unfortunatelly used/reserved for member attributes/modifiers in various syntaxes

example:

```py
@annotation
def func():
    pass
```

I have not final proposal for a suitable scope, but `meta.type.php` was introduced for type annotations in PHP recently. Maybe we want to use something like that for all kinds of type hints - also in function parameter lists and return types.

Reply:
Yeah `meta.variables.parameters` is non-sense.  Simply `meta.variable.annotation.python` is a better fit here.

We should not restrict to type-hints though.  These still can be used for meta-programming.